### PR TITLE
Add links to source code from analytics events documentation

### PR DESCRIPTION
**Pull request by zachmargolis**
Related to: https://github.com/18F/identity-idp/pull/7208 

**Example:**

It adds links to [source definition](https://github.com/18f/identity-idp/blob/dfca5e2a/app/services/analytics_events.rb#L299), [source usages](https://github.com/search?q=email_language_updated+repo%3A18f%2Fidentity-idp+extension%3Arb&type=code)

**Screenshot:**

Screen Shot 2022-10-24 at 5 08 58 PM

### CODE REVIEW

File: assets/js/components/heading.tsx [hunk 1]
```diff
@@ -15,11 +16,12 @@ export function Heading({
   level,
   children,
   id = urlify(children),
+  className,
 }: HeadingProps) {
```
Comment (aduth) [inline assets/js/components/heading.tsx:20]: Could also consider forwarding along any valid HTML props.

```suggestion
  ...props,
}: HeadingProps & JSX.HTMLAttributes<HTMLHeadingElement>) {
```

File: assets/js/github-urls.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  return `https://github.com/${repo}/blob/${ref}/${file}${
+    line ? `#L${line}` : ""
+  }`;
+}
+
+export function searchURL({
+  needle,
+  repo,
+  extension,
+  path,
```
Comment (aduth) [inline assets/js/github-urls.tsx:21]: We don't ever pass this property. Should we? (e.g. `path: "app"`)

**Review by aduth — APPROVED** [review]

File: assets/js/github-urls.tsx (hunk 2 above)
Comment (zachmargolis) [reply assets/js/github-urls.tsx:21]: I tried it out (specifically `app`) to exclude spec usages from search results, and I wasn't in love with it. But we could put it back if you think the results are better

File: assets/js/components/heading.tsx (hunk 1 above)
Comment (zachmargolis) [reply assets/js/components/heading.tsx:20]: Ok! And we'd want to remove those from the definition of HeadingProps then, right?

Comment (aduth) [reply assets/js/components/heading.tsx:20]: Yeah there's a few more revisions that would need to be done in addition to that suggestion (also importing the `JSX` type and applying the props to `TagName`). The only thing we'd need to remove from the prop type is `className`.

File: assets/js/analytics-events.tsx [hunk 3]
```diff
@@ -101,11 +121,44 @@ function Event({ event }: { event: AnalyticsEvent }) {
...
+            Source definition
+          </a>
+          ,{" "}
+          <a
+            className="usa-link"
+            href={searchURL({
+              needle: methodName,
+              repo: "18f/identity-idp",
+              extension: "rb",
+              type: "code",
```
Comment (zachmargolis) [inline assets/js/analytics-events.tsx:155]:
```suggestion
              type: "code",
              path: "app",
```

File: assets/js/github-urls.tsx (hunk 2 above)
Comment (zachmargolis) [reply assets/js/github-urls.tsx:21]: Updated in 265f764ea81e644666ad2db0420c5248456f4ba4

File: assets/js/components/heading.tsx (hunk 1 above)
Comment (aduth) [reply assets/js/components/heading.tsx:20]: y'know, in retrospect, I think this is fine as-is, if you want to just leave it.

Comment (zachmargolis) [reply assets/js/components/heading.tsx:20]: LOL ok will leave it

**Pull request merged**